pystac/item.py

```diff
@@ -483,6 +487,17 @@ def from_dict(
         if root:
             item.set_root(root)
 
+        if (
+            ItemVersionExtension.has_extension(item)
+            and ItemVersionExtension(item).deprecated
+        ):
+            warnings.warn(
+                f"The item '{item.id}' is deprecated.",
+                DeprecatedWarning,
+            )
+            # Asset deprecation checks can be added once supported by the
+            # version extension.
```

Comment:
Same as for the collection, I think it would be more "Pythonic" to just `ext` and catch the exception.

Reply:
Yep, done.